apps/managedidentity/managedidentity.go

```diff
@@ -178,8 +209,24 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
+func retry(attempts int, delay time.Duration, c ops.HTTPClient, req *http.Request) (*http.Response, error) {
+	resp, err := c.Do(req)
+	if err == nil && !slices.Contains(retryStatusCodes, resp.StatusCode) {
+		return resp, nil // Success
+	}
+	if attempts-1 < 1 {
+		return resp, nil
+	}
+	time.Sleep(delay)
+	return retry(attempts-1, delay, c, req)
```

Comment:
You probably need to rewind the request body before retrying. Make sure you have a test verifying retries have the same content as the initial attempt